R2API/UnlockableAPI.cs

```diff
@@ -145,44 +142,26 @@ public static class UnlockableAPI {
 
         [R2APISubmoduleInit(Stage = InitStage.SetHooks)]
         internal static void SetHooks() {
-            IL.RoR2.AchievementManager.CollectAchievementDefs += AddCustomAchievements;
+            SaferAchievementManager.OnCollectAchievementDefs += AddOurDefs;
         }
 
-        [R2APISubmoduleInit(Stage = InitStage.UnsetHooks)]
-        internal static void UnsetHooks() {
-            IL.RoR2.AchievementManager.CollectAchievementDefs -= AddCustomAchievements;
-        }
+        private static void AddOurDefs(List<string> identifiers, Dictionary<string, AchievementDef> stringToAchievementDef, List<AchievementDef> achievementDefs) {
+            for (var i = 0; i < Achievements.Count; i++) {
```

Comment:
Any reason to prefer a for loop vs a foreach?

Reply:
Not sure why it was written like this, just changed it